core/src/test/scala/org/apache/spark/scheduler/TaskResultGetterSuite.scala

```diff
@@ -119,5 +124,47 @@ class TaskResultGetterSuite extends SparkFunSuite with BeforeAndAfter with Local
     // Make sure two tasks were run (one failed one, and a second retried one).
     assert(scheduler.nextTaskId.get() === 2)
   }
+
+  // SPARK-11195
+  // Make sure we are using the context classloader when deserializing failed TaskResults instead of
+  // the Spark classloader.
+  test("failed task deserialized with the correct classloader") {
+    // compile a small jar containing an exception that will be thrown on an executor.
+    val tempDir = Utils.createTempDir()
+    val srcDir = new File(tempDir, "repro/")
+    srcDir.mkdirs()
+    val excSource = new JavaSourceFromString(new File(srcDir, "MyException").getAbsolutePath,
+      """package repro;
+        |
+        |public class MyException extends Exception {
+        |}
+      """.stripMargin)
+    val excFile = TestUtils.createCompiledClass("MyException", srcDir, excSource, Seq.empty)
+    val jarFile = new File(tempDir, "testJar-%s.jar".format(System.currentTimeMillis()))
+    TestUtils.createJar(Seq(excFile), jarFile, directoryPrefix = Some("repro"))
+
+    // load the exception from the jar
+    val loader = new MutableURLClassLoader(new Array[URL](0), Thread.currentThread.getContextClassLoader)
+    loader.addURL(jarFile.toURI.toURL)
+    Thread.currentThread().setContextClassLoader(loader)
+    val excClass: Class[_] = Class.forName("repro.MyException", false, loader)
+
+    sc = new SparkContext("local", "test", conf)
```

Comment:
I think we need to add a comment to say we need to use `local` in order to make this test work. Otherwise, that jar will not be loaded at the executor side.